src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +258,76 @@ private EmailWrapper generateSubmissionConfirmationEmail(
 
     }
 
+    private EmailWrapper generateLinkRecoveryEmailForUser(String userEmail) {
+        Instant endTime = Instant.now();
+        Instant startTime = endTime.minus(Duration.ofDays(180));
+        String subject = EmailType.FEEDBACK_ACCESS_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String teammateHomePageLink = Config.APP_URL;
+        String emailBody;
+        String studentName = null;
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllFeedbackSessionsWithinTimeRange(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
+            CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+            StudentAttributes student = studentsLogic.getStudentForEmail(course.getId(), userEmail);
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "(Feedback session is not yet open)";
+                String reportUrlHtml = "(Feedback session is not yet published)";
+
+                if (session.isOpened()) {
+                    String submitUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSION_SUBMISSION_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    submitUrlHtml = "<a href=\"" + submitUrl + "\">" + submitUrl + "</a>";
+                } else if (session.isClosed()) {
+                    submitUrlHtml = "(Feedback session is closed)";
```

Comment:
yeah I think that is for reportUrl, submitUrl is for submit response only

Reply:
Hi junming, sorry previously i missed out this comment. Not all sessions are published. So what if the session is closed and not published? we should still send the students the email containing the access link, which is the `studentFeedbackSubmissionEditPage` link.

In light of this, do you mind writing a test containing test cases for different kinds of sessions i.e. open + unpublished, open + published, closed + unpublished, closed + published to see if you get the desired email links? since right now without frontend this is the only way to check whether you send out the correct email. 

sorry for the trouble :(

Reply:
`what if the session is closed and not published?` 
Do you mean in this case we should send email for user to view their historical response?

Reply:
For a student whos not registered on TEAMMATES, once he lost the emails he would lose every single session that he attended/should attend. By my interpretation of this issue, we aim to provide the student with the access to all his sessions in the past 180 days, regardless of whether he needs to submit a response or not...I know some links might be "unnecessary" in the sense that the student might not care about his past responses if he cannot change them. But what if some unregistered students want to know about their past responses? (Yup I know they can retrieve all their links by "registering on TEAMMATES", but not all of them know about it/want to do it.). Let's ask for some help here @wkurniawan07

Reply:
> By my interpretation of this issue, we aim to provide the student with the access to all his sessions in the past 180 days, regardless of whether he needs to submit a response or not

That's correct.

Reply:
I have just added tests for all combinations. open + unpublished, open + published, closed + unpublished, closed + published. Please check, Thank you!